index.js

```diff
@@ -116,33 +97,203 @@ function getConstructorName(errorLike) {
+
+  // Handle `errLike` being omitted and `errMsgMatcher` being passed as the
+  // second argument.
+  if (errLikeType === 'string' || errLikeType === 'regexp') {
+    if (errMsgMatcherType !== 'undefined' && errMsgMatcherType !== 'null') {
+      throw TypeError('errMsgMatcher must be null or undefined when errLike' +
+                      ' is a string or regular expression');
+    }
+    errMsgMatcher = errLike;
+    errMsgMatcherType = errLikeType;
+    errLikeType = 'null';
+  }
+
+  if (errLikeType === 'undefined' || errLikeType === 'null') {
+    errLike = Error;
+    errLikeType = type(errLike).toLowerCase();
+  }
+
+  var isErrLikeInstanceOfError = errLike instanceof Error;
+  var isErrLikeErrorConstructor = errLike === Error ||
+    (errLikeType === 'function' && errLike.prototype instanceof Error);
+  if (!isErrLikeInstanceOfError && !isErrLikeErrorConstructor) {
+    throw TypeError('errLike must be an Error constructor or instance,' +
+                    ' string, regular expression, or criteria object');
+  }
+
+  if (errMsgMatcherType !== 'undefined' && errMsgMatcher !== 'null' &&
+      errMsgMatcherType !== 'string' && errMsgMatcherType !== 'regexp') {
+    throw TypeError('errMsgMatcher must be a string or regular expression');
+  }
+
+  if (isErrLikeInstanceOfError && errMsgMatcherType !== 'undefined' &&
+      errMsgMatcherType !== 'null') {
+    throw TypeError('errMsgMatcher must be null or undefined when errLike is' +
+                    ' an Error instance');
+  }
+
+  return {
+    id: criteriaMarker,
+    errLike: errLike,
+    errLikeType: errLikeType,
+    errMsgMatcher: errMsgMatcher,
+    errMsgMatcherType: errMsgMatcherType,
+    isErrLikeInstanceOfError: isErrLikeInstanceOfError,
+    isErrLikeErrorConstructor: isErrLikeErrorConstructor,
+  };
+}
+
+/**
+ * ### .describeExpectedError([errLike, [errMsgMatcher]])
+ *
+ * Return a string describing what kind of `Error` instance is expected based
+ * on criteria defined by `errLike` and/or `errMsgMatcher`.
+ *
+ * This function allows `errLike` to be omitted and `errMsgMatcher` to be
+ * passed in as the first argument. In other words, this function signature is
+ * also supported: `.describeExpectedError([errMsgMatcher])`.
+ *
+ * If `errLike` is omitted or is explicitly `null` or `undefined`, then it
+ * defaults to the built-in `Error` constructor.
+ *
+ * See `.checkError` for more info on `errLike` and `errMsgMatcher`.
+ *
+ * @param {Error|ErrorConstructor} [errLike=Error]
+ * @param {String|RegExp} [errMsgMatcher]
+ * @returns String
+ */
+function describeExpectedError(errLike, errMsgMatcher) {
+  var criteria = createCriteria(errLike, errMsgMatcher);
+  if (criteria.errLike instanceof Error) {
+    // Mimic `util.inspect`.
+    return '[' + criteria.errLike.toString() + ']';
+  }
+
+  var desc = '';
+  var constructor = getConstructorName(criteria.errLike);
+  var first = constructor.charAt(0).toLowerCase();
+  var vowels = [ 'a', 'e', 'i', 'o', 'u' ];
```

Comment:
Maybe a premature optimisation, but this array doesn't need to be created each time. Could have it as a const